### PR TITLE
Parser for Acknowledgments 

### PR DESCRIPTION
**Pull request by tantikristanti**
A branch containing a model for parsing acknowledgment section both for a raw string text and for the Pdf files. Result from the DESIR Code sprint 2.

### CODE REVIEW

File: grobid-trainer/resources/dataset/acknowledgment/evaluation/test.tei.xml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (lfoppiano) [inline grobid-trainer/resources/dataset/acknowledgment/evaluation/test.tei.xml]: If you use the n-fold validation you could move all the data in the training, which will be partitioned before running the test/evaluation sets and then you can use the whole dataset to train, with a bit more data.

File: grobid-trainer/src/main/java/org/grobid/trainer/AcknowledgmentTrainer.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+     * Command line execution.
+     *
+     * @param args Command line arguments.
+     * @throws Exception
+     */
+    public static void main(String[] args) throws Exception {
+        GrobidProperties.getInstance();
+        AcknowledgmentTrainer trainer = new AcknowledgmentTrainer();
+        AbstractTrainer.runTraining(trainer);
+        AbstractTrainer.runEvaluation(trainer);
```
Comment (lfoppiano) [inline grobid-trainer/src/main/java/org/grobid/trainer/AcknowledgmentTrainer.java]: See comment before. Might not need the evaluation if you use the n-fold validation

**Review by lfoppiano — CHANGES_REQUESTED** [review]
Thank you for this PR. I passed through it and added some comments. Please have a look at them. 

Also why do you need the clearNLP dictionary `data/clearNLP/dictionary-1.3.1.zip`? 

Thank you

File: grobid-trainer/src/main/java/org/grobid/trainer/AcknowledgmentTrainer.java (hunk 2 above)
Comment (tantikristanti) [reply grobid-trainer/src/main/java/org/grobid/trainer/AcknowledgmentTrainer.java]: OK, thanks, I will look at it. 
Do you have any documentation to put me on the right path?
Here:
https://grobid.readthedocs.io/en/latest/Training-the-models-of-Grobid/#n-folds-cross-evaluation, will be enough? or 
here 
https://grobid-quantities.readthedocs.io/en/latest/training.html ?

File: grobid-home/models/date/model.wapiti [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#mdl#2#263
```
[the remark this answers is not in the document]
Comment (tantikristanti) [reply grobid-home/models/date/model.wapiti]: sorry for this, I retrained several models just as examples but I forgot to return them back before the commit..

File: grobid-home/models/date/backup/model.wapiti [hunk 4]
```diff
@@ -0,0 +1,6921 @@
+#mdl#2#263
```
[the remark this answers is not in the document]
Comment (tantikristanti) [reply grobid-home/models/date/backup/model.wapiti]: same reason, I was careless not selecting or deleting the files before commiting

File: grobid-core/src/main/java/org/grobid/core/engines/FullTextParser.java [hunk 5]
```diff
@@ -307,6 +301,10 @@ else if (config.getConsolidateCitations() == 2)
 				//System.out.println(rese);
 			}
 
+			// TODO : acknowledgment like annexes
+            //documentBodyParts = doc.getDocumentPart(SegmentationLabels.ACKNOWLEDGEMENT);
```
[the remark this answers is not in the document]
Comment (tantikristanti) [reply grobid-core/src/main/java/org/grobid/core/engines/FullTextParser.java]: not needed, it was only comments during discussions and reminders for me

File: grobid-core/src/main/java/org/grobid/core/engines/DateParser.java [hunk 6]
```diff
@@ -397,6 +397,7 @@ else if (i == ll - 1) {
                     }*/
 
                 String output = writeField(s1, lastTag0, s2, "<day>", "<day>", addSpace, 0);
+                System.out.println(output);
```
[the remark this answers is not in the document]
Comment (tantikristanti) [reply grobid-core/src/main/java/org/grobid/core/engines/DateParser.java]: yes, I have compared it with the master branch

File: .travis.yml [hunk 7]
```diff
@@ -1,5 +1,6 @@
 language: java
 sudo: true
+dist: trusty
```
[the remark this answers is not in the document]
Comment (tantikristanti) [reply .travis.yml]: yes you are right, I won't need it. Actually I had bugs which made the test in TestFullTextParser failed. I fixed the bugs.

File: grobid-trainer/src/main/java/org/grobid/trainer/AcknowledgmentTrainer.java (hunk 2 above)
Comment (lfoppiano) [reply grobid-trainer/src/main/java/org/grobid/trainer/AcknowledgmentTrainer.java]: yes, the first link 

in AbstractTrainer there is a method called `evaluateNFold` or something like that (I'm trying to remember)

File: grobid-core/src/main/java/org/grobid/core/engines/FullTextParser.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (lfoppiano) [reply grobid-core/src/main/java/org/grobid/core/engines/FullTextParser.java]: Ok, just remember to cleanup before sending the PR ;-)

File: grobid-home/models/acknowledgment/TrainResult.txt [hunk 8]
```diff
@@ -1,401 +0,0 @@
-(base) MP16-PRO:grobid tkristan$ ./gradlew train_acknowledgment
```
Comment (lfoppiano) [inline grobid-home/models/acknowledgment/TrainResult.txt]: this seems the same as the evaluation result - should remove one of the two - ideally this one

**Review by lfoppiano — CHANGES_REQUESTED** [review]

File: grobid-core/src/main/java/org/grobid/core/data/Acknow.java [hunk 9]
```diff
@@ -0,0 +1,33 @@
+package org.grobid.core.data;
+
```
Comment (lfoppiano) [inline grobid-core/src/main/java/org/grobid/core/data/Acknow.java]: What is this class for? Could you add a little description?

File: grobid-core/src/test/java/org/grobid/core/engines/AcknowledgmentParserTest.java [hunk 10]
```diff
@@ -31,13 +32,53 @@ public void setUp() throws Exception {
...
+
         final Acknowledgment acknowledgment = output.get(0);
         assertThat(acknowledgment.getFundingAgency(), is("the Deutsche Forschungsgemeinschaft"));
         assertThat(acknowledgment.getProjectName(), is("the SFB 649 \" Economic Risk \""));
 
         //System.out.println(output.get(0).getFundingAgency());
     }
+
+    @Test
+    public void processing_akwnowledgment2_shouldWork() throws Exception {
```
Comment (lfoppiano) [inline grobid-core/src/test/java/org/grobid/core/engines/AcknowledgmentParserTest.java]: This test is not testing anything. Please use assertions ;-)

**Review by lfoppiano — CHANGES_REQUESTED** [review]

**Comment by kermitt2** [comment]
Si finally covered with PR #1046, with some re-design, re-annotation and more training data.

**Pull request closed without merging**